lib/Collection.js

```diff
@@ -0,0 +1,197 @@
+ */
+'use strict';
+
+const axios = require('axios');
+const bedrock = require('bedrock');
+const {EdvClient, EdvDocument} = require('edv-client');
+const https = require('https');
+const instances = require('./instances');
+const {util: {BedrockError}} = bedrock;
+
+const DEFAULT_HEADERS = {Accept: 'application/ld+json, application/json'};
+
+module.exports = class Collection {
+  constructor({type, instance, controllerKey, capability}) {
+    this.type = type;
+    this.instance = instance;
+    this.capability = capability;
+    this.controllerKey = controllerKey;
+  }
+
+  // FIXME: bikeshed name
+  static async getInstance({type, instance}) {
+    if(typeof instance === 'string') {
+      ({instance} = await instances.get({id: instance}));
+    }
+    const controllerKey = await instances.getControllerKey({id: instance.id});
+    const capability = await instances.getCapability(
+      {instance, referenceId: `${instance.id}-edv-configuration`});
+    return new Collection({type, instance, controllerKey, capability});
+  }
+
+  async create({item, meta}) {
+    if(item.type !== this.type) {
+      throw new TypeError(`"item.type" (${item.type}) must be "${this.type}".`);
+    }
+    const {controllerKey, instance, capability} = this;
+    const edvDoc = await getEdvDocument(
+      {controllerKey, instance, capability});
+    const id = await EdvClient.generateId();
+    await edvDoc.write({doc: {id, content: item, meta}});
+  }
+
+  async get({id, token}) {
+    const {controllerKey, instance, capability} = this;
+    let results;
+    if(id) {
+      results = await findDocuments({controllerKey, instance, id, capability});
+    } else if(token) {
+      results = await findDocuments({
+        controllerKey,
+        instance,
+        equals: {'content.type': this.type, 'meta.token.id': token},
+        capability
+      });
+    } else {
+      throw new TypeError('"id" or "token" must be given.');
+    }
+    if(results.length > 0) {
+      return results[0];
+    }
+    return null;
+  }
+
+  async getAll() {
+    const {controllerKey, instance, capability} = this;
+    const results = await findDocuments(
+      {controllerKey, instance, type: this.type, capability});
+    return results;
+  }
+
+  async update({item, meta, sequence}) {
+    if(item.type !== this.type) {
+      throw new TypeError(`"item.type" (${item.type}) must be "${this.type}".`);
+    }
+    const {controllerKey, instance, capability} = this;
+    const existing = await this.get({id: item.id});
+    const edvDoc = await getEdvDocument(
+      {controllerKey, instance, id: existing.id, capability});
+    const doc = await edvDoc.read();
+    if(sequence !== undefined && sequence !== doc.sequence) {
+      const err = new BedrockError(
+        'Could not update item; a conflict occurred.',
+        'InvalidStateError');
+      throw err;
+    }
+    const updatedDoc = {
+      ...doc
+    };
+    if(item) {
```

Comment:
Is it possible for `item` to be falsey here?  `item.type` is being accessed on L74.

Reply:
Removed -- `item` must be given now.